src/node.cc

```diff
@@ -2297,6 +2297,84 @@ void Init(std::vector<std::string>* argv,
+                exec_argv->end(),
+                "--experimental_report") == exec_argv->end()) {
+    if (std::find(exec_argv->begin(), exec_argv->end(), "--report-directory") !=
+            exec_argv->end() ||
+        std::find(exec_argv->begin(), exec_argv->end(), "--report_directory") !=
+            exec_argv->end()) {
+      fprintf(stderr,
+              "--report-directory option is valid only when "
+              "--experimental-report is set.\n");
+      exit(9);
+    }
+
+    if (std::find(exec_argv->begin(), exec_argv->end(), "--report-filename") !=
+            exec_argv->end() ||
+        std::find(exec_argv->begin(), exec_argv->end(), "--report_filename") !=
+            exec_argv->end()) {
+      fprintf(stderr,
+              "--report-filename option is valid only when "
+              "--experimental-report is set.\n");
+      exit(9);
+    }
+    if (std::find(exec_argv->begin(),
+                  exec_argv->end(),
+                  "--report-on-fatalerror") != exec_argv->end() ||
+        std::find(exec_argv->begin(),
+                  exec_argv->end(),
+                  "--report_on_fatalerror") != exec_argv->end()) {
+      fprintf(stderr,
+              "--report-on-fatalerror option is valid only when "
+              "--experimental-report is set.\n");
+      exit(9);
+    }
+    if (std::find(exec_argv->begin(), exec_argv->end(), "--report-on-signal") !=
+            exec_argv->end() ||
+        std::find(exec_argv->begin(), exec_argv->end(), "--report_on_signal") !=
+            exec_argv->end()) {
+      fprintf(stderr,
+              "--report-on-signal option is valid only when "
+              "--experimental-report is set.\n");
+      exit(9);
+    }
+    if (std::find(exec_argv->begin(), exec_argv->end(), "--report-signal") !=
+            exec_argv->end() ||
+        std::find(exec_argv->begin(), exec_argv->end(), "--report_signal") !=
+            exec_argv->end()) {
+      fprintf(stderr,
+              "--report-signal option is valid only when --experimental-report "
+              "is set.\n");
+      exit(9);
+    }
+    if (std::find(exec_argv->begin(),
+                  exec_argv->end(),
+                  "--report-uncaught-exception") != exec_argv->end() ||
+        std::find(exec_argv->begin(),
+                  exec_argv->end(),
+                  "--report_uncaught_exception") != exec_argv->end()) {
+      fprintf(stderr,
+              "--report-uncaught-exception option is valid only when "
+              "--experimental-report is set.\n");
+      exit(9);
+    }
+    if (std::find(exec_argv->begin(), exec_argv->end(), "--report-verbose") !=
+            exec_argv->end() ||
+        std::find(exec_argv->begin(), exec_argv->end(), "--report_verbose") !=
+            exec_argv->end()) {
+      fprintf(stderr,
+              "--report-verbose option is valid only when "
+              "--experimental-report is set.\n");
+      exit(9);
+    }
+  }
+#endif
+
```

Comment:
I believe we could better here; either by having a re-usable function (say `Mandate` on the lines of `Implies` and `ImpliesNot`) in the parser that could perform this at a post-parse stage, however I was unable to develop its semantics right, so came up with this one.

Reply:
I think this might be handled in PerProcessOptions:CheckOptions instead.Something like

if (!experimental_report) {
  if (report_uncaught_exception) {
    errors->push_back("--report-on-fatalerror option is valid only when --experimental-report is set.");
  }

  if (report-on-signal) {
    errors->push_back("--report-on-signal option is valid only when --experimental-report is set.");
  }

etc.. for the rest of the options.
}

Reply:
thanks @mhdawson - I was able to follow this, pushed a commit to that effect.